src/analysis/analysis-service.js

```diff
@@ -94,13 +94,37 @@ AnalysisService.prototype._removeAnalsyisFromIndex = function (analysis) {
 
 /**
  * Return a list with all the analyses contained in the given collections.
+ * 
+ * @example
+ * We have the following analyses:  (a0->a1->a2), (b0->a2)
+ * This method will give us: (a0->a1->a2), (b0->a2)
  */
 AnalysisService.getAnalysisList = function (layersCollection, dataviewsCollection) {
   var layerAnalyses = _getAnalysesFromLayers(layersCollection);
   var dataviewsAnalyses = _getAnalysesFromDataviews(dataviewsCollection);
   return layerAnalyses.concat(dataviewsAnalyses);
 };
 
+/**
+ * Return all the analysis nodes without duplicates.
+ * The analyses are obtained from the layers and dataviews collections.
+ * 
+ * @example
+ * We have the following analyses:  (a0->a1->a2), (b0->a2)
+ * This method will give us: (a0->a1->a1), (a1->a2), (a2), (b0->a2)
+ */
+AnalysisService.getUniqueAnalysesNodes = function (layersCollection, dataviewsCollection) {
```

Comment:
Do we want to add some tests for this method?

Reply:
`the current test suite already covers this` :P

Reply:
Perhaps indirectly. I'm referring to adding unit tests for this new function...